### PR TITLE
Disable pipeline SA

### PR DESCRIPTION
**Pull request by Roming22**
The PR has 3 commits. This is because some of the changes are temporary and will need to be rolled back in the future.

### CODE REVIEW

File: operator/gitops/argocd/pipeline-service/openshift-pipelines/allow-argocd-to-manage.yaml [hunk 1]
```diff
@@ -14,6 +14,16 @@ rules:
...
+  - apiGroups:
+      - security.openshift.io
+    resources:
+      - securitycontextconstraints
+    verbs:
+      - get
+      - list
+      - patch
+      - create
+      - delete
```
Comment (gabemontero) [inline operator/gitops/argocd/pipeline-service/openshift-pipelines/allow-argocd-to-manage.yaml:26]: do we really need delete ?   In the wrong hands could be destructive

File: operator/gitops/argocd/pipeline-service/openshift-pipelines/pipelines-scc.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+metadata:
+  name: pipelines-scc
+allowHostDirVolumePlugin: false
+allowHostIPC: false
+allowHostNetwork: false
+allowHostPID: false
+allowHostPorts: false
+allowPrivilegeEscalation: false
+allowPrivilegedContainer: false
+allowedCapabilities:
```
Comment (gabemontero) [inline operator/gitops/argocd/pipeline-service/openshift-pipelines/pipelines-scc.yaml:12]: I mentioned this in @ramessesii2 PR on this as well .... some comments on why we need the deltas from the restricted SCC would help here

Comment (gabemontero) [reply operator/gitops/argocd/pipeline-service/openshift-pipelines/pipelines-scc.yaml:12]: I suspect they are related to building with buildah ... did you perhaps just copy what was being done from the pipeline SA, or is this a delta of what was done for the pipeline SA ?

File: operator/gitops/argocd/pipeline-service/openshift-pipelines/allow-argocd-to-manage.yaml (hunk 1 above)
Comment (gabemontero) [reply operator/gitops/argocd/pipeline-service/openshift-pipelines/allow-argocd-to-manage.yaml:26]: I'm guessing that perhaps the openshift pipelines operator had this as part of doing uninstall?

If so, I guess ok .... but if somebody ever hacked a pod with this permission ....

Comment (Roming22) [reply operator/gitops/argocd/pipeline-service/openshift-pipelines/allow-argocd-to-manage.yaml:26]: Those permissions are needed to create the pipelines-scc resource. It is a temporary solution and will be removed in the future.

File: operator/gitops/argocd/pipeline-service/openshift-pipelines/pipelines-scc.yaml (hunk 2 above)
Comment (Roming22) [reply operator/gitops/argocd/pipeline-service/openshift-pipelines/pipelines-scc.yaml:12]: Those permissions were provided by @Michkov and ensures that build-service can run its pipelines.
We are temporarily responsible for those permissions, and we expect those changes to be reverted (which is why they are bundled in their own commit).

Comment (ramessesii2) [reply operator/gitops/argocd/pipeline-service/openshift-pipelines/pipelines-scc.yaml:12]: I also tried comparing the one created automatically by Pipelines to the one provided by Michal and they seem the same to me. It's just we'll be creating them instead on Pipelines for now.

File: operator/gitops/argocd/pipeline-service/openshift-pipelines/allow-argocd-to-manage.yaml (hunk 1 above)
Comment (gabemontero) [reply operator/gitops/argocd/pipeline-service/openshift-pipelines/allow-argocd-to-manage.yaml:26]: ok

File: operator/gitops/argocd/pipeline-service/openshift-pipelines/pipelines-scc.yaml (hunk 2 above)
Comment (gabemontero) [reply operator/gitops/argocd/pipeline-service/openshift-pipelines/pipelines-scc.yaml:12]: ok so yes to my question (buildah related == what Michal would provide) and they are temporary

**Review by gabemontero — APPROVED** [review]

**Pull request merged**